fastlane_bot/events/pools/solidly_v2.py

```diff
@@ -88,11 +88,13 @@ def update_from_contract(
         """
         reserve_balance = contract.caller.getReserves()
 
-        try:
+        if hasattr(contract.caller, "factory"):
             factory_address = contract.caller.factory()
-        except Exception:
-            # Velocimeter does not expose factory function - call voter to get an address that is the same for all Velcoimeter pools
+        elif hasattr(contract.caller, "voter"):
             factory_address = contract.caller.voter()
+        else:
+            slot = contract.w3.eth.get_storage_at(contract.address, 11)
```

Comment:
can you please add a comment here what this does? as a reader I can kind follow the factory / voter things but this is just black magic

Reply:
Fixed